internal/host/options.go

```diff
@@ -66,3 +69,19 @@ func WithStartPageAfterItem(item pagination.Item) Option {
 		return nil
 	}
 }
+
+// WithReaderWriter is used to share the same database reader
+// and writer when executing sql within a transaction.
+func WithReaderWriter(r db.Reader, w db.Writer) Option {
+	return func(o *options) error {
+		if r == nil {
+			return errors.New("reader cannot be nil")
+		}
+		if w == nil {
+			return errors.New("writer cannot be nil")
+		}
+		o.WithReader = r
+		o.WithWriter = w
+		return nil
+	}
+}
```

Comment:
I think we have tests for this option type, do you mind adding some tests for these new options to this PR?

Reply:
I added some tests.